pkg/install/resources.go

```diff
@@ -132,3 +139,81 @@ func VolumeSnapshotLocation(namespace, provider string, config map[string]string
 		},
 	}
 }
+
+func Secret(namespace string, data []byte) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: objectMeta(namespace, "cloud-credentials"),
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		Data: map[string][]byte{
+			"cloud": data,
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+}
+
+func appendUnstructured(list *unstructured.UnstructuredList, obj runtime.Object) error {
+	u, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&obj)
+	if err != nil {
+		return err
+	}
+	list.Items = append(list.Items, unstructured.Unstructured{Object: u})
+	return nil
+}
+
+// AllResources returns a list of all resources necessary to install Velero, in the appropriate order, into a Kubernetes cluster.
+// Items are unstructured, since there are different data types returned.
+func AllResources(namespace, image, backupStorageProviderName, bucketName, prefix string, bslConfig, vslConfig map[string]string, secretData []byte, restoreOnly bool) (*unstructured.UnstructuredList, error) {
+	resources := new(unstructured.UnstructuredList)
+	// Set the GVK so that the serialization framework outputs the list properly
+	resources.SetGroupVersionKind(schema.GroupVersionKind{Group: "", Version: "v1", Kind: "List"})
+
+	for _, crd := range CRDs() {
+		unstructuredCRD, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&crd)
+		// Remove the status field, as it's added with null values by the code, but is rejected by server-side validation
+		delete(unstructuredCRD, "status")
```

Comment:
we should probably do this for all of the resources -- I noticed that if you do `-o yaml`, then you get empty statuses for some of the other types. Would be best to not have that.

Reply:
I had this in the `appendUnstructured` function; I'll move it there again.